src/System.Diagnostics.Process/src/System/Diagnostics/ProcessStartInfo.cs

```diff
@@ -102,62 +80,23 @@ public bool CreateNoWindow
 
                     // Manual use of IDictionaryEnumerator instead of foreach to avoid DictionaryEntry box allocations.
                     IDictionaryEnumerator e = envVars.GetEnumerator();
-                    try
+                    while (e.MoveNext())
                     {
-                        while (e.MoveNext())
-                        {
-                            DictionaryEntry entry = e.Entry;
-                            _environmentVariables.Add((string)entry.Key, (string)entry.Value);
-                        }
-                    }
-                    finally
-                    {
-                        (e as IDisposable)?.Dispose();
```

Comment:
In short, yes.  If you foreach over a Hashtable, you allocate an enumerator object, then you box a DictionaryEntry for every access to Current, and if your keys/values are structs, you're boxing those simply by adding them to the Hashtable (which is non-generic and stores objects).  If you can switch to a `Dictionary<TKey, TValue>` instead, then foreach will pick-up the pattern-based struct enumerator, so you won't allocate an enumerator object, and that enumerator has a strongly-typed-KeyValuePair Current, so you won't box on each iteration, and when storing struct values, you avoid boxing when adding the keys/values into the dictionary.